qa/standalone/osd/osd-rep-recov-eio.sh

```diff
@@ -171,6 +174,86 @@ function TEST_rados_get_with_eio() {
     delete_pool $poolname
 }
 
+function TEST_rados_repair_warning() {
+    local dir=$1
+    local OBJS=$(expr $warnings + 1)
+
+    setup_osds 4 || return 1
+
+    local poolname=pool-rep
+    create_pool $poolname 1 1 || return 1
+    wait_for_clean || return 1
+
+    local poolname=pool-rep
+    local obj-base=obj-warn-
+    local inject=eio
+
+   for i in $(seq 1 $OBJS)
+    do
+      rados_put $dir $poolname ${objbase}-$i || return 1
+      inject_$inject rep data $poolname ${objbase}-$i $dir 0 || return 1
+      rados_get $dir $poolname ${objbase}-$i || return 1
+    done
+    local pgid=$(get_pg $poolname ${objbase}-1)
+
+    local object_osds=($(get_osds $poolname ${objbase}-1))
+    local primary=${object_osds[0]}
+    local bad_peer=${object_osds[1]}
+
+    COUNT=$(ceph pg $pgid query | jq '.info.stats.stat_sum.num_objects_repaired')
+    test "$COUNT" = "$OBJS" || return 1
+    flush_pg_stats
+    COUNT=$(ceph pg dump --format=json-pretty | jq ".pg_map.osd_stats_sum.num_shards_repaired")
+    test "$COUNT" = "$OBJS" || return 1
+
+    ceph health | grep -q "Too many repaired reads on 1 OSDs" || return 1
+    ceph health detail | grep -q "osd.$primary had $OBJS reads repaired" || return 1
+
+    ceph health mute OSD_TOO_MANY_REPAIRS
```

Comment:
Nautilus does not have the ability to mute health warnings (https://github.com/ceph/ceph/pull/29422). This is causing failures like https://pulpito.ceph.com/yuriw-2020-07-21_22:37:25-rados-wip-yuri5-testing-2020-07-17-2032-nautilus-distro-basic-smithi/5248207/

Reply:
@neha-ojha Just remove lines 212 - 216 to remove the mute test?

Reply:
@neha-ojha In Nautilus if we backport, this HEALTH_ERR condition can't be cleared presumably until the OSD is replaced.

Reply:
@dzafman yeah, that's the way to go, @smithfarm can you test this change locally?

Reply:
>  can you test this change locally?

@neha-ojha Not easily. Also, I'm concerned about David's other comment: "In Nautilus if we backport, this HEALTH_ERR condition can't be cleared presumably until the OSD is replaced."

Reply:
I've create a commit that adds a new tell command for OSDs that can set the num_shards_repaired.  This is usually used without an argument to clear it.  For testing you can specify a count.  The test case for this feature uses the new tell command in lieu of mute.